tutorials/using-cloud-vpn-with-cisco-asr/index.md

```diff
@@ -0,0 +1,893 @@
+
+### Advanced VPN configurations
+
+#### Configure VPN redundancy
+
+![alt_text](https://storage.googleapis.com/gcp-community/tutorials/using-cloud-vpn-with-cisco-asr/GCP-Cisco-ASR-Topology-Redundant.jpg)
+
+Using redundant tunnels ensures continuous availability in the case of a tunnel fails.
+
+If a Cloud VPN tunnel goes down, it restarts automatically. If an entire virtual
+device fails, Cloud VPN automatically instantiates a new one with the same
+configuration, so you don't need to build two Cloud VPN gateways. The new
+gateway and tunnel connect automatically. For hardware appliances such as Cisco
+ASR it is recommended that you deploy atleast 2 ASRs and create VPN tunnels to
+GCP from each for redundancy purposes.
+
+The VPN redundancy configuration example is built based on the IPsec tunnel and
+BGP configuration illustrated above.
+
+##### Cisco ASR
+
+Cisco IOS BGP prefer the path with the highest `LOCAL-PREF`, the BGP routes are
+set with a value of 100 by default, by setting the `LOCAL-PREF` to 200 for the
+routes received from Tunnel1, BGP will choose Tunnel1 as the preferred VPN
+tunnel to the GCP, in the event of Tunnel 1 failure, BGP will reroute the
+traffic to Tunnel2.
+
+    crypto ikev2 keyring VPN_SCALE_TEST_KEY
+     peer GCP1
+     address 104.196.200.68
+     pre-shared-key MySharedSecret
+     peer GCP2
+     address 35.186.108.199
+     pre-shared-key MySharedSecret
+    !
+    interface Tunnel1
+     description VPN tunnel to the east coast DC
+     ip address 169.254.0.2 255.255.255.252
+     ip mtu 1400
+     ip tcp adjust-mss 1360
+     tunnel source TenGigabitEthernet0/0/0
+     tunnel mode ipsec ipv4
+     tunnel destination 104.196.200.68
+     tunnel protection ipsec profile VPN_SCALE_TEST_VTI
+    !
+    interface Tunnel2
+     description VPN tunnel to the west coast DC
+     ip address 169.254.0.6 255.255.255.252
+     ip mtu 1400
+     ip tcp adjust-mss 1360
+     tunnel source TenGigabitEthernet0/0/0
+     tunnel mode ipsec ipv4
+     tunnel destination 35.186.108.199
+     tunnel protection ipsec profile VPN_SCALE_TEST_VTI_2
+    !
+
+###### Dynamic Routing
+
+    router bgp 65001
+     bgp log-neighbor-changes
+     neighbor 169.254.0.1 description BGP session over Tunnel1
+     neighbor 169.254.0.1 remote-as 65002
+     neighbor 169.254.0.1 timers 20 60 60
+     neighbor 169.254.0.5 description BGP session over Tunnel2
+     neighbor 169.254.0.5 remote-as 65002
+     neighbor 169.254.0.5 timers 20 60 60
+     !
+     address-family ipv4
+      network 10.0.0.0
+      neighbor 169.254.0.1 activate
+      neighbor 169.254.0.1 route-map LP200 in
+      neighbor 169.254.0.5 activate
+     exit-address-family
+    !
+    route-map LP2000 permit 10
+     set local-preference 2000
+
+To ensure symmetry in your traffic flow, you can configure MED to influence the
+inbound traffic from GCP for the same tunnel you are sending outbound traffic
+to. Note that lower the MED, higher the preference.
```

Comment:
I think what you're saying is that to use the same tunnel for both inbound and outbound traffic, you need to set preference for one tunnel. To do that, set a lower MED value on your router and on Cloud Router for that tunnel (compared to the other tunnel). The lower the value the higher the priority. Is that right? Can we make that clearer?

Reply:
no changes are required on Cloud router. Cloud router will honor the MED value received from the Cisco router and route traffic accordingly. Let me know if it is still not clear.